app-api/database/versions/1824436b3cf9_dataset_jobs.py

```diff
@@ -0,0 +1,47 @@
+"""dataset jobs
+
+Revision ID: 1824436b3cf9
+Revises: b8f0b197b931
+Create Date: 2025-02-11 09:30:11.754804
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = "1824436b3cf9"
+down_revision: Union[str, None] = "b8f0b197b931"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "dataset_jobs",
+        sa.Column("id", sa.UUID(), nullable=False),
+        sa.Column("user_id", sa.UUID(), nullable=False),
+        sa.Column("dataset_id", sa.UUID(), nullable=False),
+        sa.Column("extra_metadata", sa.JSON(), nullable=False),
+        sa.Column("secret_metadata", sa.JSON(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["dataset_id"],
+            ["datasets.id"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["user_id"],
+            ["users.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
```

Comment:
This new DB entity tracks currently running dataset jobs (the job ID that this has with the inference service).